integration/volume_create_test.go

```diff
@@ -0,0 +1,141 @@
+package integration
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/http/httputil"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/sclevine/spec"
+	"github.com/stretchr/testify/require"
+)
+
+var _ = suite("compute/volume/create", func(t *testing.T, when spec.G, it spec.S) {
+	var (
+		expect   *require.Assertions
+		cmd      *exec.Cmd
+		baseArgs []string
+	)
+
+	it.Before(func() {
+		expect = require.New(t)
+
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			switch req.URL.Path {
+			case "/v2/volumes":
+				auth := req.Header.Get("Authorization")
+				if auth != "Bearer some-magic-token" {
+					w.WriteHeader(http.StatusUnauthorized)
+					return
+				}
+
+				if req.Method != "POST" {
+					w.WriteHeader(http.StatusTeapot)
```

Comment:
You should probably use the correct method for this, in case we ever want to test that aspect of the API.
```suggestion
					w.WriteHeader(http.StatusMethodNotAllowed)
```